pal/src/tlsio_openssl_compact.c

```diff
@@ -458,7 +458,9 @@ static void dowork_read(TLS_IO_INSTANCE* tls_io_instance)
         //
         // Pump all of the bytes currently available out
         rcv_bytes = SSL_read(tls_io_instance->ssl, buffer, sizeof(buffer));
-        while (rcv_bytes > 0)
+        // After 100 SSL_read calls, break this while loop to avoid an infinite condition. 
+        int idx = 0; 
+        while (rcv_bytes > 0 && idx++ < 100)
```

Comment:
So this is for when SSL_read keeps returning > 0 forever, right?

Wondering if 100 is too much.
If we run our code on a esp32, what is the maximum number of times SSL_read returns > 0 in a row?
Mark has been working with esp32, maybe he could run a little test on this.

Reply:
Correct. We can run tests to validate the number of loops that might be necessary.